### PR TITLE
fix: commits get sent despite denying prompt

### PR DESCRIPTION
**Pull request by tobealive**
This PR deals with the bug that commits get sent no matter what a user is answering to the `"Are you sure you want to commit?"` confirmation prompt. Especially when accidentally entering a `commit --amend` and it gets commited desipte canceling, it gets a bit annoying because the state before the commit cannot be restored so easily.

It looks like the reason for this behavior is a somewhat messy conditional statement in the commit editor. 

~~Also, a visual feedback on cancellation might be nice - as it is done with the other events. But there doesn't seem to be a way to get a return value that can be used others with the other notification messages without doing some architetural changes. So I'm helping myself out with an autocmd.~~

edited: 2022/11/04

### CODE REVIEW

**Comment by TimUntersberger** [comment]
> Also, a visual feedback on cancellation would be nice - as it is done with the other events. But there doesn't seem to be a way to get a return value that can be used others with the other notification messages without doing some architetural changes. So I'm helping myself out with an autocmd.

There shouldn't be a need for visual feedback. If you say no it just doesn't do anything, so there is no async part to it.

File: lua/neogit/buffers/commit_editor/init.lua [hunk 1]
```diff
@@ -40,24 +40,24 @@ function M:open()
     modifiable = true,
     readonly = false,
     autocmds = {
-      ["BufWritePost"] = function()
-        written = true
```
Comment (TimUntersberger) [inline lua/neogit/buffers/commit_editor/init.lua:44]: why did you remove the written condition?

File: lua/neogit/buffers/commit_editor/init.lua [hunk 2]
```diff
@@ -40,24 +40,24 @@ function M:open()
...
-            config.values.disable_commit_confirmation
-            or input.get_confirmation("Are you sure you want to commit?")
-          then
-            vim.cmd([[
-              silent g/^#/d
-              silent w!
-            ]])
-          end
+        if
+          not config.values.disable_commit_confirmation
```
Comment (TimUntersberger) [inline lua/neogit/buffers/commit_editor/init.lua:45]: this condition doesn't do the same thing. if you disable the commit confirmation you never enter into this condition

Comment (tobealive) [reply lua/neogit/buffers/commit_editor/init.lua:45]: > > Also, a visual feedback on cancellation would be nice - as it is done with the other events. But there doesn't seem to be a way to get a return value that can be used others with the other notification messages without doing some architetural changes. So I'm helping myself out with an autocmd.
> 
> There shouldn't be a need for visual feedback. If you say no it just doesn't do anything, so there is no async part to it.

Perfectly fine. Tell me if you like it via rebase or as commit, and I'll remove it. Or if you prefer go ahead push the changes and add yourself as co-author.

**Comment by tobealive** [comment]
> > Also, a visual feedback on cancellation would be nice - as it is done with the other events. But there doesn't seem to be a way to get a return value that can be used others with the other notification messages without doing some architetural changes. So I'm helping myself out with an autocmd.
> 
> There shouldn't be a need for visual feedback. If you say no it just doesn't do anything, so there is no async part to it.

That's just fine. Tell me if you like it via rebase or as commit, and I'll remove it. Or if you prefer go ahead push the changes and add yourself as co-author.

File: lua/neogit/buffers/commit_editor/init.lua (hunk 2 above)
Comment (tobealive) [reply lua/neogit/buffers/commit_editor/init.lua:45]: yes that's right. but isn't it the goal that you won't get the confirmation message if you disable it?

File: lua/neogit/buffers/commit_editor/init.lua (hunk 1 above)
Comment (tobealive) [reply lua/neogit/buffers/commit_editor/init.lua:44]: As far as I could see, we could achieve the desired functionality with a negating guard clause condition. Which is less nested with less code. Also, I've seen the crook in here that this is the reason why commits are always send. As it would always be true due to the hard-coded save commands. 

And the rebase editor also has none: https://github.com/TimUntersberger/neogit/blob/3323380e4a110ec3bfa4d2a681f4a9634a239e87/lua/neogit/buffers/rebase_editor/init.lua#L27-L30 so I'll burned it.

Alternatively it should work to use the original with less modification and additionally add an else statement to return the function early.

**Comment by tobealive** [comment]
@TimUntersberger 

> There shouldn't be a need for visual feedback. If you say no it just doesn't do anything, so there is no async part to it.

I removed the notification when a commit gets canceled.

> this condition doesn't do the same thing. if you disable the commit confirmation you never enter into this condition

Checking the functionality, I think we should be fine. Please clarify any other changes that need to be made, so I can understand and carry out the desired result if I am missing something.

**Comment by tobealive** [comment]
Should be fixed with the recent updates. Therefore closing.
